### PR TITLE
Minmax property fix

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: include/inviwo/core/properties/minmaxproperty.h [hunk 1]
```diff
@@ -310,8 +308,10 @@ template <typename T>
 MinMaxProperty<T>& MinMaxProperty<T>::setRangeNormalized(const range_type& newRange) {
     dvec2 val = this->get();
 
-    val = (val - static_cast<double>(range_.value.x)) /
-          (static_cast<double>(range_.value.y) - static_cast<double>(range_.value.x));
+    const auto nomalizedValue =
```
Comment (r-englund) [inline include/inviwo/core/properties/minmaxproperty.h:309]: unused?

Comment (petersteneteg) [reply include/inviwo/core/properties/minmaxproperty.h:309]: ahh forgot to finish that one...

**Review by r-englund — APPROVED** [review]

File: include/inviwo/core/properties/minmaxproperty.h [hunk 2]
```diff
@@ -272,29 +272,27 @@ MinMaxProperty<T>& MinMaxProperty<T>::setMinSeparation(const T& newMinSeparation
 }
 
 template <typename T>
-MinMaxProperty<T>& MinMaxProperty<T>::setRange(const range_type& value) {
-    const auto newRange = clamp({glm::min(value.x, value.y), glm::max(value.x, value.y)});
+MinMaxProperty<T>& MinMaxProperty<T>::setRange(const range_type& newRange) {
+    if (range_.update({glm::min(newRange.x, newRange.y), glm::max(newRange.x, newRange.y)})) {
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/minmaxproperty.h:276]: Perhaps add a comment why this is necessary or should be done. Is it not an error if the new range is flipped?

File: include/inviwo/core/properties/minmaxproperty.h [hunk 3]
```diff
@@ -272,29 +272,27 @@ MinMaxProperty<T>& MinMaxProperty<T>::setMinSeparation(const T& newMinSeparation
...
-void MinMaxProperty<T>::set(const range_type& value, const range_type& range, const T& increment,
-                            const T& minSep) {
-
-    const auto newRange = clamp({glm::min(range.x, range.y), glm::max(range.x, range.y)});
+void MinMaxProperty<T>::set(const range_type& newValue, const range_type& newRange,
+                            const T& newIncrement, const T& newMinSep) {
 
     bool modified = false;
-    modified |= range_.update(newRange);
+    modified |= range_.update({glm::min(newRange.x, newRange.y), glm::max(newRange.x, newRange.y)});
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/minmaxproperty.h:290]: Again, is this not an input error?

**Review by ResearchDaniel — DISMISSED** [review]
I’m not sure if incorrect range input should be corrected. Throw invalid argument exception?

File: include/inviwo/core/properties/minmaxproperty.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/minmaxproperty.h:276]: This is just to be defensive that the new min is less then the new max we have always done this...

File: include/inviwo/core/properties/minmaxproperty.h (hunk 3 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/minmaxproperty.h:290]: this is how we have used to do it, if you want to throw go ahead and add that...

File: include/inviwo/core/properties/minmaxproperty.h (hunk 1 above)
Comment (martinfalk) [reply include/inviwo/core/properties/minmaxproperty.h:309]: could this be moved into the if statement?

**Review by martinfalk — APPROVED** [review]

File: include/inviwo/core/properties/minmaxproperty.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/minmaxproperty.h:309]: No since it depends on the old range...

**Pull request merged**